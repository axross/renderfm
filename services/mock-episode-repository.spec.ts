import { MockEpisodeRepository } from "~/services/mock-episode-repository.server";

describe("MockEpisodeRepository", () => {
  test("#getAll() resolves with a constant episode list", async () => {
    const episodeRepository = new MockEpisodeRepository();

    await expect(episodeRepository.getAll()).resolves.toMatchSnapshot();
  });

  test("#getById() resolves with a constant episode by the given id", async () => {
    const episodeRepository = new MockEpisodeRepository();

    await expect(episodeRepository.getById("1")).resolves.toMatchSnapshot();
  });

  test("#getById() resolves with null if there's no match episode with the given id", async () => {
    const episodeRepository = new MockEpisodeRepository();

    await expect(episodeRepository.getById("404")).resolves.toBeNull();
  });
});
