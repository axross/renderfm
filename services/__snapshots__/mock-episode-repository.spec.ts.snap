// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MockEpisodeRepository #getAll() resolves with a constant episode list 1`] = `
Array [
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "1",
    "publishedAt": "2021-06-06T21:07:37.567Z",
    "summaryMarkdown": "Lesch and Sonsの [#310 Extended asynchronous system engine](https://occaecati.io) を聴いて[@reiciendis](http://magnam.name)と[@repellat](https://blanditiis.net)がいろいろ話しました。後編です。",
    "summaryPlainText": "Reiciendis sed sed quaerat a fugiat vitae, id laborum sapiente nihil maiores qui asperiores non quaerat commodi deserunt doloribus vitae non cupiditate dicta consectetur sed ullam beatae ullam non sed non, necessitatibus et, nihil, numquam esse et unde voluptate aliquid occaecati nulla voluptate.",
    "timeline": Array [
      Array [
        "Profit-focused executive core",
        79,
      ],
      Array [
        "Versatile national time-frame",
        133,
      ],
      Array [
        "Extended asynchronous system engine",
        161,
      ],
      Array [
        "Synergized client-server architecture",
        244,
      ],
      Array [
        "Business-focused stable framework",
        292,
      ],
      Array [
        "Advanced transitional support",
        449,
      ],
      Array [
        "Cloned human-resource knowledge base",
        966,
      ],
      Array [
        "Customizable client-driven encoding",
        1591,
      ],
      Array [
        "De-engineered full-range extranet",
        1624,
      ],
      Array [
        "Re-engineered 24/7 paradigm",
        2203,
      ],
      Array [
        "Reactive motivating data-warehouse",
        2296,
      ],
      Array [
        "Orchestration encompassing middleware",
        2355,
      ],
      Array [
        "Persevering encompassing alliance",
        2403,
      ],
      Array [
        "Organized contextually-based function",
        2533,
      ],
      Array [
        "Multi-layered zero administration system engine",
        2825,
      ],
      Array [
        "Exclusive fresh-thinking workforce",
        3041,
      ],
      Array [
        "Profit-focused coherent application",
        3147,
      ],
      Array [
        "Synergistic responsive service-desk",
        3155,
      ],
    ],
    "title": "Innovative background attitude",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "4",
    "publishedAt": "2021-09-14T02:58:03.120Z",
    "summaryMarkdown": "Bogan - Danielの [#66 Reactive local implementation](http://consequatur.net) を聴いて[@repellat](https://in.net)と[@sed](https://quaerat.name)がいろいろ話しました。後編です。",
    "summaryPlainText": "Unde aut est voluptate quasi quae error, asperiores est quia, id consequuntur aliquid.",
    "timeline": Array [
      Array [
        "Balanced radical archive",
        1267,
      ],
      Array [
        "Enhanced responsive software",
        1358,
      ],
      Array [
        "Synergized 4th generation success",
        1569,
      ],
      Array [
        "Re-engineered upward-trending standardization",
        1625,
      ],
      Array [
        "Streamlined impactful alliance",
        1743,
      ],
      Array [
        "Enhanced responsive software",
        1761,
      ],
      Array [
        "Optimized contextually-based toolset",
        1830,
      ],
      Array [
        "Reactive local implementation",
        2184,
      ],
      Array [
        "Virtual background paradigm",
        2642,
      ],
      Array [
        "User-centric optimizing implementation",
        2806,
      ],
      Array [
        "Networked mobile moratorium",
        2824,
      ],
      Array [
        "Persevering encompassing middleware",
        2906,
      ],
      Array [
        "Configurable global help-desk",
        2929,
      ],
    ],
    "title": "Optimized contextually-based toolset",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "13",
    "publishedAt": "2021-03-11T10:49:19.220Z",
    "summaryMarkdown": "Brown LLCの [#349 Multi-lateral zero defect throughput](http://sunt.org) を聴いて[@numquam](https://sequi.com)と[@nihil](https://sunt.dev)がいろいろ話しました。後編です。",
    "summaryPlainText": "Sequi aut sapiente in consectetur sed possimus tenetur, quaerat magnam nemo, consequatur beatae tenetur id repellat occaecati quos asperiores est ipsum consequatur.",
    "timeline": Array [
      Array [
        "Multi-layered zero administration system engine",
        1560,
      ],
      Array [
        "Seamless bi-directional capacity",
        1688,
      ],
      Array [
        "Secured holistic architecture",
        2850,
      ],
      Array [
        "Secured holistic architecture",
        2851,
      ],
    ],
    "title": "Focused asymmetric contingency",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "21",
    "publishedAt": "2022-01-26T21:38:10.282Z",
    "summaryMarkdown": "Hegmann Incの [#76 Distributed heuristic archive](http://doloribus.io) を聴いて[@quasi](http://error.net)と[@voluptate](http://id.info)がいろいろ話しました。後編です。",
    "summaryPlainText": "Consectetur commodi dolores et ipsum, nemo enim beatae laborum consectetur doloribus voluptate, voluptatem, sit deserunt vel sit quas id in et consectetur est voluptatem quos.",
    "timeline": Array [
      Array [
        "Robust foreground synergy",
        270,
      ],
      Array [
        "Balanced radical archive",
        368,
      ],
      Array [
        "Fundamental multi-tasking standardization",
        455,
      ],
      Array [
        "Upgradable cohesive knowledge user",
        480,
      ],
      Array [
        "Switchable next generation intranet",
        557,
      ],
      Array [
        "Implemented well-modulated task-force",
        682,
      ],
      Array [
        "Enhanced responsive software",
        896,
      ],
      Array [
        "User-friendly impactful utilisation",
        1716,
      ],
      Array [
        "Advanced transitional support",
        1808,
      ],
      Array [
        "Multi-channelled zero tolerance core",
        1880,
      ],
      Array [
        "Enterprise-wide directional orchestration",
        2241,
      ],
      Array [
        "De-engineered needs-based hardware",
        2265,
      ],
      Array [
        "Inverse contextually-based portal",
        2378,
      ],
      Array [
        "Synergistic responsive service-desk",
        2435,
      ],
      Array [
        "Synchronised 24 hour emulation",
        2558,
      ],
      Array [
        "Networked mobile moratorium",
        2956,
      ],
    ],
    "title": "Compatible fresh-thinking success",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "40",
    "publishedAt": "2021-04-24T08:16:15.213Z",
    "summaryMarkdown": "Orn, Spencer and Kiehnの [#436 Upgradable cohesive knowledge user](http://nihil.info) を聴いて[@magnam](http://excepturi.name)と[@numquam](https://nihil.io)がいろいろ話しました。後編です。",
    "summaryPlainText": "Commodi et cum tenetur in, fugiat est dolores dicta vel non, repellat, a at at deserunt nostrum cupiditate possimus enim.",
    "timeline": Array [
      Array [
        "Right-sized client-driven firmware",
        19,
      ],
      Array [
        "Ameliorated asynchronous pricing structure",
        171,
      ],
      Array [
        "Reduced multi-tasking encoding",
        924,
      ],
      Array [
        "Secured methodical frame",
        1052,
      ],
      Array [
        "User-centric executive knowledge user",
        1569,
      ],
      Array [
        "Virtual background paradigm",
        2183,
      ],
      Array [
        "Multi-channelled zero tolerance core",
        2208,
      ],
      Array [
        "Organized contextually-based function",
        2248,
      ],
    ],
    "title": "Re-engineered 24/7 paradigm",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "42",
    "publishedAt": "2021-06-08T02:56:24.559Z",
    "summaryMarkdown": "Walker - Ziemeの [#472 Reactive 4th generation info-mediaries](http://neque.com) を聴いて[@beatae](https://quasi.com)と[@commodi](https://labore.biz)がいろいろ話しました。後編です。",
    "summaryPlainText": "Rerum consequuntur exercitationem exercitationem sed.",
    "timeline": Array [
      Array [
        "Secured holistic architecture",
        545,
      ],
      Array [
        "Proactive client-server access",
        1068,
      ],
      Array [
        "Synergized client-server architecture",
        1126,
      ],
      Array [
        "Enterprise-wide directional orchestration",
        1204,
      ],
      Array [
        "Secured holistic architecture",
        2682,
      ],
      Array [
        "Innovative background attitude",
        2805,
      ],
    ],
    "title": "Expanded didactic methodology",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "44",
    "publishedAt": "2021-02-09T23:04:16.551Z",
    "summaryMarkdown": "Hahn - Welchの [#469 Streamlined executive Graphic Interface](http://cupiditate.info) を聴いて[@sequi](https://ullam.biz)と[@sequi](http://consequuntur.net)がいろいろ話しました。後編です。",
    "summaryPlainText": "Facilis possimus beatae nihil voluptatibus id repellat esse consectetur vitae nemo fugiat doloribus.",
    "timeline": Array [
      Array [
        "Adaptive asymmetric infrastructure",
        24,
      ],
      Array [
        "Face to face 6th generation complexity",
        264,
      ],
      Array [
        "Fundamental multi-tasking standardization",
        424,
      ],
      Array [
        "Reactive 4th generation info-mediaries",
        476,
      ],
      Array [
        "Distributed heuristic archive",
        661,
      ],
      Array [
        "Exclusive fresh-thinking workforce",
        775,
      ],
      Array [
        "Mandatory regional complexity",
        844,
      ],
      Array [
        "Re-contextualized grid-enabled portal",
        884,
      ],
      Array [
        "Multi-layered zero administration system engine",
        1093,
      ],
      Array [
        "Upgradable stable strategy",
        1173,
      ],
      Array [
        "Integrated regional open architecture",
        1178,
      ],
      Array [
        "Robust heuristic installation",
        1342,
      ],
      Array [
        "Re-engineered upward-trending standardization",
        1723,
      ],
      Array [
        "Multi-channelled zero tolerance core",
        1797,
      ],
      Array [
        "Enterprise-wide 24 hour info-mediaries",
        2121,
      ],
      Array [
        "Stand-alone needs-based pricing structure",
        2136,
      ],
      Array [
        "Cloned human-resource knowledge base",
        2359,
      ],
    ],
    "title": "Organized maximized firmware",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "57",
    "publishedAt": "2021-05-25T23:23:24.962Z",
    "summaryMarkdown": "Swaniawski and Sonsの [#2 Distributed heuristic archive](https://tenetur.org) を聴いて[@repellat](https://deserunt.info)と[@maiores](http://reiciendis.dev)がいろいろ話しました。後編です。",
    "summaryPlainText": "Deserunt esse dolores deserunt non blanditiis excepturi reiciendis, qui vitae vitae ullam est nulla, exercitationem maiores sed.",
    "timeline": Array [
      Array [
        "Stand-alone next generation task-force",
        449,
      ],
      Array [
        "Extended asynchronous system engine",
        1197,
      ],
    ],
    "title": "Exclusive fresh-thinking workforce",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "67",
    "publishedAt": "2021-05-13T16:34:16.850Z",
    "summaryMarkdown": "Marquardt - Runolfsdottirの [#208 Robust foreground synergy](https://id.net) を聴いて[@vitae](https://quia.org)と[@ducimus](http://ducimus.info)がいろいろ話しました。後編です。",
    "summaryPlainText": "Sapiente sed qui sunt fugiat hic aut asperiores quas facilis fugiat occaecati dicta sit numquam, in quia, enim quas rerum eos doloribus sit dolores blanditiis ullam consequuntur neque quas quas ducimus qui labore, numquam consequuntur.",
    "timeline": Array [
      Array [
        "Decentralized human-resource system engine",
        483,
      ],
      Array [
        "Versatile national time-frame",
        885,
      ],
      Array [
        "Synergized client-server architecture",
        1641,
      ],
      Array [
        "Right-sized client-driven firmware",
        1884,
      ],
      Array [
        "Streamlined executive Graphic Interface",
        2009,
      ],
      Array [
        "Multi-layered zero administration system engine",
        2170,
      ],
      Array [
        "Stand-alone next generation task-force",
        2521,
      ],
      Array [
        "Streamlined executive Graphic Interface",
        2522,
      ],
      Array [
        "Switchable upward-trending array",
        2617,
      ],
      Array [
        "Organized contextually-based function",
        2732,
      ],
      Array [
        "Stand-alone needs-based pricing structure",
        3028,
      ],
      Array [
        "Networked exuding monitoring",
        3175,
      ],
      Array [
        "Reduced multi-tasking encoding",
        3177,
      ],
    ],
    "title": "Reactive motivating data-warehouse",
  },
  Object {
    "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
    "duration": 3186.703673469388,
    "id": "100",
    "publishedAt": "2021-06-21T17:26:49.910Z",
    "summaryMarkdown": "Champlin - Gaylordの [#89 Monitored bottom-line productivity](http://magnam.info) を聴いて[@eos](http://ipsum.dev)と[@sequi](http://quos.io)がいろいろ話しました。後編です。",
    "summaryPlainText": "Omnis occaecati necessitatibus voluptatem ducimus consectetur tenetur doloribus sequi, id sapiente, nostrum et nihil tenetur.",
    "timeline": Array [
      Array [
        "De-engineered full-range extranet",
        177,
      ],
      Array [
        "Adaptive asymmetric infrastructure",
        674,
      ],
      Array [
        "Innovative background attitude",
        1209,
      ],
      Array [
        "Polarised global open system",
        1345,
      ],
      Array [
        "Ameliorated multimedia groupware",
        1367,
      ],
      Array [
        "Business-focused stable framework",
        1433,
      ],
      Array [
        "Reduced regional frame",
        1490,
      ],
      Array [
        "Robust foreground synergy",
        2333,
      ],
      Array [
        "Robust heuristic installation",
        2920,
      ],
      Array [
        "Business-focused stable framework",
        2920,
      ],
    ],
    "title": "Synergized client-server architecture",
  },
]
`;

exports[`MockEpisodeRepository #getById() resolves with a constant episode by the given id 1`] = `
Object {
  "audioUrl": "https://podcast-pitpa-enginner.cdn.pitpa.jp/renderfm/episodes/001_Jcfeg.mp3",
  "duration": 3186.703673469388,
  "id": "1",
  "publishedAt": "2021-06-06T21:07:37.567Z",
  "summaryMarkdown": "Lesch and Sonsの [#310 Extended asynchronous system engine](https://occaecati.io) を聴いて[@reiciendis](http://magnam.name)と[@repellat](https://blanditiis.net)がいろいろ話しました。後編です。",
  "summaryPlainText": "Reiciendis sed sed quaerat a fugiat vitae, id laborum sapiente nihil maiores qui asperiores non quaerat commodi deserunt doloribus vitae non cupiditate dicta consectetur sed ullam beatae ullam non sed non, necessitatibus et, nihil, numquam esse et unde voluptate aliquid occaecati nulla voluptate.",
  "timeline": Array [
    Array [
      "Profit-focused executive core",
      79,
    ],
    Array [
      "Versatile national time-frame",
      133,
    ],
    Array [
      "Extended asynchronous system engine",
      161,
    ],
    Array [
      "Synergized client-server architecture",
      244,
    ],
    Array [
      "Business-focused stable framework",
      292,
    ],
    Array [
      "Advanced transitional support",
      449,
    ],
    Array [
      "Cloned human-resource knowledge base",
      966,
    ],
    Array [
      "Customizable client-driven encoding",
      1591,
    ],
    Array [
      "De-engineered full-range extranet",
      1624,
    ],
    Array [
      "Re-engineered 24/7 paradigm",
      2203,
    ],
    Array [
      "Reactive motivating data-warehouse",
      2296,
    ],
    Array [
      "Orchestration encompassing middleware",
      2355,
    ],
    Array [
      "Persevering encompassing alliance",
      2403,
    ],
    Array [
      "Organized contextually-based function",
      2533,
    ],
    Array [
      "Multi-layered zero administration system engine",
      2825,
    ],
    Array [
      "Exclusive fresh-thinking workforce",
      3041,
    ],
    Array [
      "Profit-focused coherent application",
      3147,
    ],
    Array [
      "Synergistic responsive service-desk",
      3155,
    ],
  ],
  "title": "Innovative background attitude",
}
`;
